raiden-ts/src/channels/epics.ts

```diff
@@ -595,21 +595,23 @@ export const channelMonitoredEpic = (
   );
 
 const makeDeposit$ = (
-  tokenContract: HumanStandardToken,
-  tokenNetworkContract: TokenNetwork,
-  sender: Address,
-  address: Address,
-  partner: Address,
+  [tokenContract, tokenNetworkContract]: [HumanStandardToken, TokenNetwork],
+  [sender, address, partner]: [Address, Address, Address],
   deposit: UInt<32> | undefined,
   channelId$: Observable<number>,
   { log }: Pick<RaidenEpicDeps, 'log'>,
+  approveNonce?: number,
 ): Observable<channelDeposit.failure> => {
   if (!deposit?.gt(Zero)) return EMPTY;
   return defer(() => tokenContract.functions.allowance(sender, tokenNetworkContract.address)).pipe(
     mergeMap((allowance) =>
       allowance.gte(deposit)
         ? of(true)
-        : from(tokenContract.functions.approve(tokenNetworkContract.address, deposit)).pipe(
+        : from(
+            tokenContract.functions.approve(tokenNetworkContract.address, deposit, {
+              nonce: approveNonce,
```

Comment:
Just to make sure: if `approveNonce` is `undefined` it gets ignored and the auto-nonce mechanism works as before? I assume so. But sometimes library implementations are special. Better safe than sorry...

Reply:
Yes, exactly, this is actually the case with the norma `channelDeposit.request`, since it shouldn't race there, so needed only here because of the concurrent `channelOpen`  call.